From: Mark Rustad <mark.d.rustad@intel.com>
Date: Wed, 28 Sep 2016 15:36:38 -0700
Subject: ixgbe: Handle previously-freed msix_entries
Patch-mainline: v4.10-rc1
Git-commit: 1fa71252778d6fe151438b009416e4ce7bc1c802
References: bsc#1024349 FATE#321240 bsc#1024355 FATE#321241

The msix_entries memory can be freed by a previous suspend or
remove, so don't crash on close when it isn't there.

Signed-off-by: Mark Rustad <mark.d.rustad@intel.com>
Tested-by: Krishneil Singh <krishneil.k.singh@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/intel/ixgbe/ixgbe_main.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
@@ -3067,6 +3067,9 @@ static void ixgbe_free_irq(struct ixgbe_
 		return;
 	}
 
+	if (!adapter->msix_entries)
+		return;
+
 	for (vector = 0; vector < adapter->num_q_vectors; vector++) {
 		struct ixgbe_q_vector *q_vector = adapter->q_vector[vector];
 		struct msix_entry *entry = &adapter->msix_entries[vector];
